src/controllers/controller.line.js

```diff
@@ -248,30 +249,44 @@ module.exports = function(Chart) {
 			var meta = me.getMeta();
 			var area = me.chart.chartArea;
 
-			// only consider points that are drawn in case the spanGaps option is ued
+			// Only consider points that are drawn in case the spanGaps option is used
 			var points = (meta.data || []).filter(function(pt) { return !pt._model.skip; });
 			var i, ilen, point, model, controlPoints;
 
-			var needToCap = me.chart.options.elements.line.capBezierPoints;
```

Comment:
It's not removed, if I've done that right. It is moved after the calls to `splineCurve` or `splineCurveMonotone`.

I also think it should be faster this way, because very few JavaScript compilers are able to remove or hoist this `needToCap` check (thus ASM jump) in the `capIfNecessary` function (call a few times in the loop). It is also a lot more difficult to avoid/not execute/not compile the loop and call.

`capControlPoint` should be quite easily inlined by most recent compilers, and the loop can be completely ignored (and even not compiled at all for very recent compilers that do branch pruning). Not that it matters much for loops on < 100 items ^^,